openmc/checkvalue.py

```diff
@@ -41,9 +41,9 @@ def check_type(name, value, expected_type, expected_iter_type=None):
         Description of value being checked
     value : object
         Object to check type of
-    expected_type : type
+    expected_type : type or Iterable of type
         type to check object against
-    expected_iter_type : type or None, optional
+    expected_iter_type : type or Iterable of type or None, optional
```

Comment:
Nice, I've thought for some time now that such extensions would be useful, but hadn't run into a use case which really needed it yet. But I can see why this PR could make use of this, as well as future improvements to the API.